### DISCUSSION TITLE
Feedback for colored AMWG HTML tables

### DISCUSSION THREAD

**Discussion by justin-richling** [NCAR/ADF#144] [Ideas]
With the suggestion from @swrneale about possibly adding color to highlight our HTML tables, I have made progress on styling the pandas data frames that generate our HTML files. This all works on the backend with python and not in editing the HTML files after.

My initial test was for a hypothetical case vs case vs obs and highlight values of either case that were closest to the obs in green and the furthest in red. I have a Jupyter notebook that can provide a quick testing ground of it and sample output HTML files here: ```/glade/work/richling/ADF/ADF_dev/notebooks/```

<img width="1058" alt="Screen Shot 2022-03-30 at 1 09 28 PM" src="https://user-images.githubusercontent.com/56696811/160912261-e7fb0064-4646-4f55-9d89-eb5e2d22b440.png">

Obviously this is not production ready yet as the ADF does not account for a three way comparison. However, the styling can be implemented for other situations like testing both case and baseline against a preferred threshold or ideal value, etc. Rich provided some examples [here](https://webext.cgd.ucar.edu/AMWG/gust_mods/F2010climo.ne30pg3_ne30pg3_mg17.TestGustGlobal_58L.002-F2010climo.ne30pg3_ne30pg3_mg17.TestGustGlobal_Control_58L.002/set14/set14.htm) and [here](https://www.ilamb.org/results/) that may be useful.

I have an example [here](https://project.cgd.ucar.edu/projects/ADF/f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_kzz3_zmtop75.hf.001_vs_f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001_1982_1990/website/html_table/amwg_table_f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_kzz3_zmtop75.hf.001_color.html) of a basic variable column highlight that has been integrated into a run of the ADF.

 I wanted to open a discussion for potential styling needs that I can test.

**Comment by brianpm** [comment]
This is already very cool, thanks, @justin-richling !!

The one thing that immediately occurs to me is that it might be better to color based on a test/threshold rather than by the binary winner/loser. For example, we might want to have N cases, and want all cases that are "acceptable" to be green and cases that "fail" to be red. Also, we are often comparing simulations for development where the values might differ but not actually be statistically significantly different compared to the reference value. In both these cases, maybe a statistical test for `color_rule` (e.g., a t-test to compare mean values) would be better. That could make `color_rule` more complicated, but maybe better than that would be to keep `color_rule` simple and provide the statistical test result as an input.

**Comment by swrneale** [comment]
This is cool. Thanks @justin-richling 
Following on from @brianpm ideas. As well as significance, I think hue shading strength would guide the eye on the metrics that are most/east skillful. But, I'm sure you're on to it already :)

**Comment by andrewgettelman** [comment]
Thanks Justin! The style is a dramatic improvement over what we have in there already. I say we can start with this and worry about tweaking it a bit later (others can also start tweaking color and hue as are suggested above: both good ideas). 

One note: do we have the 'short' case names available in the ADF? (e.g. 'case1' 'case2' in Justin's table). If not, we should make sure we add those: we will need them for titles and the table headers. E.g., we need to be able to have 'Kzz3' and not 'f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_kzz3_zmtop75.hf.001'

**Reply by cecilehannay** [reply]
We could have a line at the top of the table with:
- the "full name" (here: 'f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_kzz3_zmtop75.hf.001')  
- and the "nickname" (here: Kzz3)

**Comment by andrewgettelman** [comment]
I am suggesting we should use nicknames throughout the ADF (especially given those full case names....)

**Reply by cecilehannay** [reply]
I fully agree but I think it is useful to have the full name somewhere and not only the nickname. 

Right now the fullname is at the top of the page. For instance:

![image](https://user-images.githubusercontent.com/9723220/160924212-123ef974-fcfa-4fbd-8114-e120d6bfe7db.png)

I am suggesting that instead of:
Case 1: f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_kzz3_zmtop75.hf.001
Case 2: f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001

we could have:
kzz3: f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_kzz3_zmtop75.hf.001
control: f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001

**Comment by cecilehannay** [comment]
@justin-richling: I like this a lot already. 

I fully agree with all suggestions above. All great ideas. 
That said, we don't need it to be perfect before we include it. You can always improve later. My 2 cents.

**Comment by justin-richling** [comment]
These are all great suggestions, thanks! 

I will keep it simple and work towards getting an Issue and PR going on the GitHub. Keep the ideas coming!
